ast_tools/passes/bool_to_bit.py

```diff
@@ -0,0 +1,74 @@
+import ast
+import typing as tp
+
+from . import Pass
+from . import PASS_ARGS_T
+
+from ast_tools.stack import SymbolTable
+
+__ALL__ = ['bool_to_bit']
+
+class AndTransformer(ast.NodeTransformer):
+    def visit_BoolOp(self, node: ast.BoolOp) -> ast.expr:
+        # Can't get more specific on return type because if
+        # len(node.values) == 1 (which it shouldn't be)
+        # then the return type is expr otherwise
+        # the return type is Union[BinOp, BoolOp]
+
+        if isinstance(node.op, ast.And):
+            values = node.values
+            assert values # should not be empty
+            expr = self.visit(values[0])
+            for v in map(self.visit, values[1:]):
+                expr = ast.BinOp(expr, ast.BitAnd(), v)
+            return expr
+        else:
+            return self.generic_visit(node)
+
+class OrTransformer(ast.NodeTransformer):
+    def visit_BoolOp(self, node: ast.BoolOp) -> ast.expr:
+        if isinstance(node.op, ast.Or):
+            values = node.values
+            assert values # should not be empty
+            expr = self.visit(values[0])
+            for v in map(self.visit, values[1:]):
+                expr = ast.BinOp(expr, ast.BitOr(), v)
+            return expr
+        else:
+            return self.generic_visit(node)
```

Comment:
Seems like these two above classes could share a parent with a class attribute set to determine the match/replace, e.g.

```
class generic:
    def visit_BoolOp(self, node: ast.BoolOp) -> ast.expr:
        if isinstance(node.op, self.matchOp):
            ....
                expr = ast.BinOp(expr, self.replaceOp(), v)
```

then the children could simply be

```
class OrTransformer(generic):
    matchOp = ast.Or
    replaceOp = ast.BitOr
```

Reply:
Alternatively a generic bool op replacer that takes a mapping of match -> replace